### PR TITLE
MultiSelect and SingleSelect

### PR DESCRIPTION
**Pull request by GabeMedrash**

### CODE REVIEW

File: src/ui/select/src/multi-select.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+const defaultProps = {
+  placeholder: 'Add/remove'
+};
+
+export default class MultiSelect extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      width: getWidestLabel(props.options, props.labelKey, props.hierarchical) + 60
```
Comment (brettvitaz) [inline src/ui/select/src/multi-select.jsx]: magic numbers
Should be a component prop or a declared constant.

File: src/ui/select/src/multi-option.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      onMouseOver={onMouseOver}
+    >
+      <input
+        type="checkbox"
+        checked={isSelected}
+        readOnly
+      />
+      <span
+        className={classnames({ [style.selected]: isSelected })}
+        style={{ marginLeft: 20 }}
```
Comment (brettvitaz) [inline src/ui/select/src/multi-option.jsx]: magic number

File: src/ui/select/src/hierarchical-option-renderer.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  // key on option that holds its label
+  labelKey: PropTypes.string
+};
+
+const HierarchicalOption = (props) => {
+  return (
+    <span
+      style={{
+        marginLeft: (props.option.level || 0) * 5,
```
Comment (brettvitaz) [inline src/ui/select/src/hierarchical-option-renderer.jsx]: magic number

File: src/ui/select/src/value.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    this.props.onRemove(this.props.value, event);
+  }
+
+  handleMouseDown(event) {
+    if (event.type === 'mousedown' && event.button !== 0) return;
+
+    if (this.props.onClick) {
+      event.stopPropagation();
+      this.props.onClick(this.props.value, event);
+      return;
```
Comment (brettvitaz) [inline src/ui/select/src/value.jsx]: unnecessary return

**Comment by GabeMedrash** [comment]
@brettvitaz -> I refactored based on your suggestions, included some tests, consolidated some components, and then rebased on top of master.

**Comment by brettvitaz** [comment]
👍

**Pull request merged**